// Copyright 2010 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Protocol buffer for the Ticl's internal state, for persistence.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package invalidation;

message TiclState {
  // The client's uniquifier.
  optional bytes uniquifier = 1;

  // The client's session token.
  optional bytes session_token = 2;

  // The highest sequence number that the client may have used.
  optional int64 sequence_number_limit = 3;
}

// An envelope containing a Ticl's internal state, along with a digest of the
// serialized representation of this state, to ensure its integrity across
// reads and writes to persistent storage.
message StateBlob {
  // The (important parts of the) Ticl's internal state.
  optional TiclState ticl_state = 1;

  // Implementation-specific message authentication code for the Ticl state.
  optional bytes authentication_code = 2;
}
